File: client/lib/core/network/api_response.dart
class ApiResponse<T> {
  const ApiResponse({
    this.data,
    this.meta = const <String, dynamic>{},
    this.error,
  });

  final T? data;
  final Map<String, dynamic> meta;
  final ApiError? error;

  factory ApiResponse.fromJson(
    Map<String, dynamic> json,
    T Function(Object? raw) fromJson,
  ) {
    return ApiResponse<T>(
      data: json.containsKey('data') ? fromJson(json['data']) : null,
      meta: (json['meta'] as Map?)?.cast<String, dynamic>() ?? const <String, dynamic>{},
      error: json['error'] == null
          ? null
          : ApiError.fromJson((json['error'] as Map).cast<String, dynamic>()),
    );
  }
}

class ApiError {
  const ApiError({
    required this.code,
    required this.message,
  });

  final String code;
  final String message;

  factory ApiError.fromJson(Map<String, dynamic> json) {
    return ApiError(
      code: json['code'] as String? ?? 'UNKNOWN',
      message: json['message'] as String? ?? 'Unknown error',
    );
  }
}

File: client-rn/src/core/network/apiResponse.ts
/**
 * 统一 API 响应包装
 * 与后端约定的 data + meta + error 格式一致
 */
export interface ApiErrorShape {
  code: string;
  message: string;
}

export class ApiResponse<T> {
  constructor(
    public readonly data: T | null,
    public readonly meta: Record<string, unknown> = {},
    public readonly error: ApiErrorShape | null = null,
  ) {}

  static fromJson<T>(
    json: Record<string, unknown>,
    mapper: (raw: unknown) => T,
  ): ApiResponse<T> {
    return new ApiResponse<T>(
      json.data != null ? mapper(json.data) : null,
      (json.meta as Record<string, unknown>) ?? {},
      json.error != null
        ? {
            code: (json.error as Record<string, unknown>).code as string ?? 'UNKNOWN',
            message: (json.error as Record<string, unknown>).message as string ?? 'Unknown error',
          }
        : null,
    );
  }
}
